src/operator/subgraph/dnnl/dnnl_transformer_qk_common.h

```diff
@@ -0,0 +1,232 @@
+        }
+        return true;
+      }
+      break;
+    case kFirstSwapAx:
+      if (raw_input_node.op() == Op::Get("SwapAxis")) {
+        if (CheckSwapAxisConditionsQK(input_node)) {
+          *status = kSecondSwapAx;
+          matched_list->push_back(&input_node);
+          return true;
+        }
+      }
+      break;
+    case kSecondSwapAx:
+      if (raw_input_node.op() == Op::Get("_npx_reshape")) {
+        // input to reshape must be first or second output from split
+        if (CheckReshapeConditionsQK(input_node, 0) || CheckReshapeConditionsQK(input_node, 1)) {
+          *status = kFirstReshape;
+          matched_list->push_back(&input_node);
+          return true;
+        }
+      }
+      break;
+    case kFirstReshape:
+      if (raw_input_node.op() == Op::Get("_npx_reshape")) {
+        if (CheckReshapeConditionsQK(input_node, 0) || CheckReshapeConditionsQK(input_node, 1)) {
+          if constexpr (qk_mode == mode::include_split) {
+            *status = kSecondReshape;
+          } else {
+            *status = kSuccess;
+          }
+          matched_list->push_back(&input_node);
+          return true;
+        }
+      }
+      break;
+    case kSecondReshape:
+      if (raw_input_node.op() == Op::Get("_split_v2") &&
+          CheckSplitConditions(*matched_list, input_node)) {
+        *status = kSuccess;
+        return true;
+      }
+      break;
+    default:
+      *status = kFail;
+      return false;
+  }
+  return false;
+}
+
+inline std::vector<BiDirectedNode*> Filter(const SelectStatusTransformerQK& status,
+                                           const std::vector<const BiDirectedNode*>& matched_list,
+                                           const std::vector<BiDirectedNode*>& candidates) {
+  if (status != kSuccess) {
+    return std::vector<BiDirectedNode*>(0);
+  } else {
+    std::vector<BiDirectedNode*> ret;
+    for (auto i : matched_list) {
+      auto non_const_i = const_cast<BiDirectedNode*>(i);
+      if (std::find(candidates.begin(), candidates.end(), non_const_i) != candidates.end()) {
+        ret.push_back(non_const_i);
+      }
+    }
+    return ret;
+  }
+}
+
+template <mode qk_mode>
+nnvm::ObjectPtr CreateSubgraphNode(const nnvm::Symbol& sym, const int subgraph_id = 0) {
+  std::string op_name;
+  if constexpr (qk_mode == mode::include_split) {
+    op_name = "_sg_onednn_selfatt_qk_split";
+  } else {
+    op_name = "_sg_onednn_selfatt_qk";
+  }
+  nnvm::ObjectPtr n = nnvm::Node::Create();
+  // This op has single output, remove duplicated.
+  auto last_node = sym.outputs[0].node;
+  nnvm::Symbol new_sym;
+  new_sym.outputs.emplace_back(last_node);
+  std::ostringstream node_name;
+
+  DFSVisit(new_sym.outputs, [&](const nnvm::ObjectPtr& node) {
+    if ((node->op() == Op::Get("_npx_reshape"))) {
+      auto const& reshape_param = nnvm::get<NumpyXReshapeParam>(node->attrs.parsed);
+      // set heads attribute - all necessary conditions are checked before
```

Comment:
Inconsistent comments - some start from capital letter, some don't; some end with the dot, some don't.

Reply:
unified